tests/flytekit/unit/interaction/test_click_types.py

```diff
@@ -270,3 +271,69 @@ class Datum:
     assert v.y == "2"
     assert v.z == {1: "one", 2: "two"}
     assert v.w == [1, 2, 3]
+
+
+def test_nested_dataclass_type():
+    from dataclasses import dataclass
+
+    @dataclass
+    class Datum:
```

Comment:
Thank you! could we also add a few more tests?
1. Another dataclass in the `Datum`
2. List of dataclass in the `Datum`
3. Flyte type in the dataclass (expect to see an error?)
4. Add a nested dataclass in this [test](https://github.com/flyteorg/flytekit/blob/1ba65beddea2a360e3a384788bdcbe6887e8605a/tests/flytekit/unit/cli/pyflyte/test_run.py#L157-L158)